src/hooks/__test__/useAuth.test.tsx

```
@@ -0,0 +1,46 @@
+import { useAuth } from '../useAuth';
+import { AuthContext } from '@/hoc/AuthProvider';
+import { ReactNode } from 'react';
+
+// Mock context value to match the actual AuthContext interface
+const mockAuthContext = {
+    user: null,
+    isLoading: false
+};
+
+// Create a wrapper component that provides the context
+const createWrapper = (contextValue: any) => {
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)